userprofile/forms.py

```diff
@@ -57,13 +57,24 @@ def __init__(self, *args, **kwargs):
 
                 if s_val['type'] == 'STRING':
                     if 'options' in s_val:
-                        self.fields[s_field] = forms.ChoiceField(
-                            choices=[
-                                (option, option)
-                                for option in s_val.get('options')
-                            ],
-                            **setting_kwargs,
-                        )
+                        if callable(s_val.get('default')):
+                            self.fields[s_field] = forms.ChoiceField(
+                                choices=[
+                                    (str(option), str(option))
```

Comment:
Something seems to be going wrong in passing the choices into the form :)
![2023-03_form_option_bug](https://user-images.githubusercontent.com/11211145/222400610-b0aa8cba-d7e3-408d-9a58-63628fef8d42.png)

Reply:
Actually It was correct, just my example function was too weird :)